From fe356951a9d45fee73036aaa72661f0319dffba3 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Tue, 21 Aug 2018 13:30:44 +0200
Subject: daemon: Fix upower not having access to udev events

PrivateNetwork=true will cut off upower from the udev netlink socket, so
it's not what we want to use, as we want those events.

See https://lists.freedesktop.org/archives/systemd-devel/2018-July/041000.html

Closes: #68
---
 src/upower.service.in | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/upower.service.in b/src/upower.service.in
index 835529a..8838655 100644
--- a/src/upower.service.in
+++ b/src/upower.service.in
@@ -17,7 +17,8 @@ ProtectHome=true
 PrivateTmp=true
 
 # Network
-PrivateNetwork=true
+# PrivateNetwork=true would block udev's netlink socket
+RestrictAddressFamilies=AF_UNIX AF_NETLINK
 
 # Execute Mappings
 MemoryDenyWriteExecute=true
-- 
cgit v1.1

